kratos/modeler/voxel_mesh_generator_modeler.cpp

```diff
@@ -0,0 +1,335 @@
+
+        if (!parameters.Has("outside_color")) {
+            parameters.AddInt("outside_color", OutsideColor);
+        }
+
+        VoxelMesherColoring::Pointer p_coloring = factory.Create(*this, parameters);
+
+        p_coloring->ValidateParameters();
+
+        std::string model_part_name = parameters["model_part_name"].GetString();
+        std::string type = parameters["type"].GetString();
+        KRATOS_INFO("Modeler") << "Applying color to " << type << " for " << model_part_name  << " model part" << std::endl;
+
+        p_coloring->Apply();
+    }
+    Timer::Stop("MeshColoring");
+}
+
+void VoxelMeshGeneratorModeler::SetStartIds(ModelPart& rTheVolumeModelPart)
+{
+    ModelPart& r_root_model_part = rTheVolumeModelPart.GetRootModelPart();
+    std::size_t start_node_proposal = r_root_model_part.NodesArray().empty() ? 1 : r_root_model_part.NodesArray().back()->Id() + 1;
+    mStartNodeId = std::max(start_node_proposal, mStartNodeId);
+
+    std::size_t start_element_proposal = r_root_model_part.ElementsArray().empty() ? 1 : r_root_model_part.ElementsArray().back()->Id() + 1;
+    mStartElementId = std::max(start_element_proposal, mStartElementId);
+
+    std::size_t start_condition_proposal = r_root_model_part.ConditionsArray().empty()? 1 :r_root_model_part.ConditionsArray().back()->Id() + 1;
+    mStartConditionId = std::max(start_condition_proposal, mStartConditionId);
+}
+
+void VoxelMeshGeneratorModeler::GenerateEntities(
+    ModelPart& rTheVolumeModelPart,
+    Parameters EntityGeneratorParameters
+    )
+{
+    Timer::Start("EntityGeneration");
+    VoxelMesherEntityGenerationFactory factory("entity generation strategy");
+
+    for(auto parameters : EntityGeneratorParameters){
+
+        auto p_generator = factory.Create(*this, parameters);
+        p_generator->ValidateParameters();
+
+        std::string model_part_name = parameters["model_part_name"].GetString();
+        std::string type = parameters["type"].GetString();
+        KRATOS_INFO("Modeler") << "Generate " << type << " in " << model_part_name  << " model part" << std::endl;
+
+        p_generator->Generate();
+    }
+    Timer::Stop("EntityGeneration");
+}
+
+ModelPart& VoxelMeshGeneratorModeler::CreateAndGetModelPart(std::string const& FullName)
+{
+        std::istringstream iss(FullName);
+        std::string name;
+        std::getline(iss, name, '.');
+        if(!mpModel->HasModelPart(name))
+            mpModel->CreateModelPart(name);
+        ModelPart* p_current_model_part = &mpModel->GetModelPart(name);
+        while (std::getline(iss, name, '.')) {
+            if(!p_current_model_part->HasSubModelPart(name)){
+                p_current_model_part->CreateSubModelPart(name);
+            }
+            p_current_model_part = &(p_current_model_part->GetSubModelPart(name));
+        }
+
+        return *p_current_model_part;
+}
+
+ModelPart& VoxelMeshGeneratorModeler::GetModelPart(std::string const& FullName)
```

Comment:
same, the Model does exactly this

Reply:
This function was not used, so I'm removing it.